[2025-10-08T04:33:11.480+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-08T04:33:11.560+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:20:14.712439+00:00 [queued]>
[2025-10-08T04:33:11.588+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:20:14.712439+00:00 [queued]>
[2025-10-08T04:33:11.593+0000] {taskinstance.py:2306} INFO - Starting attempt 4 of 4
[2025-10-08T04:33:11.652+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): run_spark_stream> on 2025-10-08 04:20:14.712439+00:00
[2025-10-08T04:33:11.685+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62: DeprecationWarning: This process (pid=744) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-08T04:33:11.690+0000] {standard_task_runner.py:64} INFO - Started process 753 to run task
[2025-10-08T04:33:11.688+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'restaurant_analytics', 'run_spark_stream', 'manual__2025-10-08T04:20:14.712439+00:00', '--job-id', '124', '--raw', '--subdir', 'DAGS_FOLDER/restaurant_dag.py', '--cfg-path', '/tmp/tmp_2w5pc4a']
[2025-10-08T04:33:11.697+0000] {standard_task_runner.py:91} INFO - Job 124: Subtask run_spark_stream
[2025-10-08T04:33:11.909+0000] {task_command.py:426} INFO - Running <TaskInstance: restaurant_analytics.run_spark_stream manual__2025-10-08T04:20:14.712439+00:00 [running]> on host c34aac175a74
[2025-10-08T04:33:12.133+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='quantruong1918@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='restaurant_analytics' AIRFLOW_CTX_TASK_ID='run_spark_stream' AIRFLOW_CTX_EXECUTION_DATE='2025-10-08T04:20:14.712439+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-10-08T04:20:14.712439+00:00'
[2025-10-08T04:33:12.137+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-08T04:33:12.139+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-08T04:33:12.142+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '/opt/spark/bin/spark-submit --master local[*] /opt/***/dags/spark_streaming.py']
[2025-10-08T04:33:12.165+0000] {subprocess.py:86} INFO - Output:
[2025-10-08T04:33:17.290+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-10-08T04:33:17.292+0000] {subprocess.py:93} INFO -   File "/opt/***/dags/spark_streaming.py", line 5, in <module>
[2025-10-08T04:33:17.295+0000] {subprocess.py:93} INFO -     from pyspark.ml import PipelineModel
[2025-10-08T04:33:17.297+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/__init__.py", line 22, in <module>
[2025-10-08T04:33:17.298+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/base.py", line 40, in <module>
[2025-10-08T04:33:17.300+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/pyspark.zip/pyspark/ml/param/__init__.py", line 32, in <module>
[2025-10-08T04:33:17.302+0000] {subprocess.py:93} INFO - ModuleNotFoundError: No module named 'numpy'
[2025-10-08T04:33:17.390+0000] {subprocess.py:93} INFO - 25/10/08 04:33:17 INFO ShutdownHookManager: Shutdown hook called
[2025-10-08T04:33:17.399+0000] {subprocess.py:93} INFO - 25/10/08 04:33:17 INFO ShutdownHookManager: Deleting directory /tmp/spark-0c7a8487-9c5b-4b90-9d09-76db0985f899
[2025-10-08T04:33:17.489+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-10-08T04:33:17.497+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-08T04:33:17.543+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-10-08T04:33:17.561+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=restaurant_analytics, task_id=run_spark_stream, run_id=manual__2025-10-08T04:20:14.712439+00:00, execution_date=20251008T042014, start_date=20251008T043311, end_date=20251008T043317
[2025-10-08T04:33:17.635+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 124 for task run_spark_stream (Bash command failed. The command returned a non-zero exit code 1.; 753)
[2025-10-08T04:33:17.694+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-10-08T04:33:17.738+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
